miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/ads/AdMobDisplayer.kt

```diff
@@ -40,81 +55,236 @@ class AdMobDisplayer(private val context: Activity) : MiniAppAdDisplayer, Corout
+
+            ad.adListener = object : AdListener() {
+                override fun onAdLoaded() {
+                    onLoaded.invoke()
                 }
 
-                ad.loadAd(AdRequest.Builder().build())
+                override fun onAdFailedToLoad(adError: LoadAdError) {
+                    interstitialAdMap.remove(adUnitId)
+                    onFailed.invoke(adError.message)
+                }
             }
+
+            ad.loadAd(AdRequest.Builder().build())
         }
     }
 
-    /** Show the interstitial ad when it is already loaded. **/
-    override fun showInterstitialAd(adUnitId: String, onClosed: () -> Unit, onFailed: (String) -> Unit) {
-        launch {
-            if (interstitialAdMap.containsKey(adUnitId) && interstitialAdMap[adUnitId]!!.isLoaded) {
-                val ad = interstitialAdMap[adUnitId]!!
-                ad.adListener = object : AdListener() {
-
-                    override fun onAdClosed() {
-                        onClosed.invoke()
+    /** Load the interstitial ad or version 20.2.0 when it is ready. **/
+    private fun loadLatestInterstitialAd(
+        adUnitId: String,
+        onLoaded: () -> Unit,
+        onFailed: (String) -> Unit
+    ) {
+        AdmobDisplayerLatest.getInstance()
+            .loadInterstitialAd(adUnitId = adUnitId) { loadStatus, errorMessage ->
+                when (loadStatus) {
+                    AdStatus.LOADED -> onLoaded.invoke()
+                    AdStatus.FAILED -> {
                         interstitialAdMap.remove(adUnitId)
+                        onFailed.invoke(errorMessage ?: "")
                     }
+                    else -> Log.d("Ad", "Something went wrong.")
                 }
+            }
+    }
 
-                ad.show()
-            } else
-                onFailed.invoke("Ad is not loaded yet")
+    /** Show the interstitial ad when it is already loaded. **/
+    override fun showInterstitialAd(
+        adUnitId: String,
+        onClosed: () -> Unit,
+        onFailed: (String) -> Unit
+    ) {
+        launch {
+            whenHasAdmobLatest { available ->
+                if (available)
+                    showLatestInterstitialAd(adUnitId, onClosed, onFailed)
+                else
+                    showOldInterstitialAd(adUnitId, onClosed, onFailed)
+            }
         }
     }
 
+    private fun showOldInterstitialAd(
+        adUnitId: String,
+        onClosed: () -> Unit,
+        onFailed: (String) -> Unit
+    ) {
+        if (interstitialAdMap.containsKey(adUnitId) && interstitialAdMap[adUnitId]!!.isLoaded) {
+            val ad = interstitialAdMap[adUnitId]!!
+            ad.adListener = object : AdListener() {
+
+                override fun onAdClosed() {
+                    onClosed.invoke()
+                    interstitialAdMap.remove(adUnitId)
+                }
+            }
+
+            ad.show()
+        } else
+            onFailed.invoke("Ad is not loaded yet")
```

Comment:
forcing unwrapping with `!!` should be changed in my opinion. I'm not very good with kotlin, but is this possible?

```suggestion
        interstitialAdMap[adUnitId]?.let { 
            if (interstitialAdMap.containsKey(adUnitId) && it.isLoaded) {
                val ad = it
                ad.adListener = object : AdListener() {
                    override fun onAdClosed() {
                        onClosed.invoke()
                        interstitialAdMap.remove(adUnitId)
                    }
                }
                ad.show()
            } else
                onFailed.invoke("Ad is not loaded yet")
        }
```

Reply:
@Climbatize noted!!